pgcli/main.py

```diff
@@ -238,8 +238,10 @@ def connect_dsn(self, dsn):
     def connect_uri(self, uri):
         uri = urlparse(uri)
         database = uri.path[1:]  # ignore the leading fwd slash
-        self.connect(database, uri.hostname, uri.username,
-                     uri.port, uri.password)
+        arguments = [database, uri.hostname, uri.username,
+                     uri.port, uri.password]
+        # unquote each URI part (they may be percent encoded)
+        self.connect(*list(map(lambda p: unquote(p) if p else p, arguments)))
```

Comment:
Sorry for slow response. This cannot be done if we use `urlparse`, in fact this results in the error that has prompted me to submit this PR:

```
pgcli/main.py:241: in connect_uri
    uri = urlparse(unquoted)

../../../.pyenv/versions/2.7.11/lib/python2.7/urlparse.py:143: in urlparse
    tuple = urlsplit(url, scheme, allow_fragments)

>               raise ValueError("Invalid IPv6 URL")
E               ValueError: Invalid IPv6 URL
```

This happens if we unquote the URI and then try to parse the strings, which now includes invalid characters according to RFC  3986.

The error message from urllib is slightly misleading, hence this PR :)